### DISCUSSION TITLE
Authenticate as either admin or user on a unified page

### DISCUSSION THREAD

**Discussion by mac501pl** [pocketbase/pocketbase#1780] [Q&A] [answered]
I'm using SvelteKit with the JS SDK. Is there a way to have an unified login page for both admins and users? Admin auth lives in a different namespace to user auth. So I need to do something like this:
```javascript
if (adminEmails.includes(email)) {
  await pocketBase.admins.authWithPassword(email, password);
} else {
  await pocketBase.collection('users').authWithPassword(email, password);
}
```

Is there a cleaner way to do that (without such check?) - I don't want to maintain a seperate admin list. Or maybe such unified login page is not a good idea in the first place?

**Comment by ganigeorgiev** [comment, marked as answer]
> Is there a way to have an unified login page for both admins and users? 

No, both are stored as different entities with different api endpoints.
If you don't want to check against a fixed `adminEmails` field, you can do a "failover" flow, aka. first try with admin and if fail then try with users:

```js
try {
  await pocketBase.admins.authWithPassword(email, password);
} catch (e) {
  await pocketBase.collection('users').authWithPassword(email, password);
}
```

> Or maybe such unified login page is not a good idea in the first place?

It depends on your use case. PocketBase admins are developer accounts - they have access to everything (including all users and collections schema) and they are usually used only to login in the PocketBase Admin UI or as part of a server-side action that requires admin-level access (eg. modifying a collection schema or the app settings).

**Reply by mac501pl** [reply]
Thanks for the reply! I guess I don't want the admin users to have ALL PocketBase privileges. So based on your comment that admin users should primairly be developer accounts, maybe it would be a good idea to just introduce a new field for the `users` table, maybe an `isAdmin` flag and grant them some bonus privileges (just for the purposes of administrative actions around the app). That way I would have both unified login and better privilege control?

**Reply by ganigeorgiev** [reply]
Yes, a flag or `role` field would work fine. Note that auth collections also have a special "Manage" API rule that can be used to allow one user (eg. "admin") to fully manage the data of another (including changing their password without providing the old one, changing the verified state, etc.).

One caveat to the flag/role/etc. field is that you have to guard in the create and update rules of your collections who can set this flag. For example to allow only PocketBase admins and app-level admins to be able to create/update a regular user to "admin" level you can use the following create/update rules:

```
(@request.data.isAdmin:isset = false || @request.auth.isAdmin = true) && ...some other rules
```
The above translates to:
- Disallow `isAdmin` submissions
- OR allow `isAdmin` submissions but require the authorized user also to have the `isAdmin` flag set

**Reply by mac501pl** [reply]
Good thing to keep in mind, thank you!
